lib/TelemetryFeed.jsm

```diff
@@ -634,33 +635,40 @@ this.TelemetryFeed = class TelemetryFeed {
    * details about endpoint schema at:
    * https://github.com/mozilla/gcp-ingestion/blob/master/docs/edge.md#postput-request
    *
+   * @param {String} namespace Namespace of the ping, such as "activity-stream" or "messaging-system".
    * @param {String} pingType  Type of the ping, such as "impression-stats".
    * @param {String} version   Endpoint version for this ping type.
    */
-  _generateStructuredIngestionEndpoint(pingType, version) {
+  _generateStructuredIngestionEndpoint(namespace, pingType, version) {
     const uuid = gUUIDGenerator.generateUUID().toString();
     // Structured Ingestion does not support the UUID generated by gUUIDGenerator,
     // because it contains leading and trailing braces. Need to trim them first.
     const docID = uuid.slice(1, -1);
-    const extension = `${pingType}/${version}/${docID}`;
+    const extension = `${namespace}/${pingType}/${version}/${docID}`;
     return `${this.structuredIngestionEndpointBase}/${extension}`;
   }
 
-  sendStructuredIngestionEvent(event_object, pingType, version) {
+  sendStructuredIngestionEvent(event_object, namespace, pingType, version) {
     if (this.telemetryEnabled && this.structuredIngestionTelemetryEnabled) {
       this.pingCentre.sendStructuredIngestionPing(
         event_object,
-        this._generateStructuredIngestionEndpoint(pingType, version),
-        { filter: ACTIVITY_STREAM_ID }
+        this._generateStructuredIngestionEndpoint(namespace, pingType, version),
+        { filter: ACTIVITY_STREAM_ID, excludeClientID: true }
       );
     }
   }
 
-  sendASRouterEvent(event_object) {
-    if (this.telemetryEnabled) {
-      this.pingCentreForASRouter.sendPing(event_object, {
-        filter: ACTIVITY_STREAM_ID,
-      });
+  sendASRouterEvent(event_object, pingType, version, excludeClientID) {
```

Comment:
I think cleaning up the `excludeClientID` would also help merging this method with `sendStructuredIngestionEvent`.

Reply:
I've also merged these two send functions as they basically did the same thing.